backend/src/main/java/com/redhat/cloud/notifications/db/repositories/BehaviorGroupRepository.java

```diff
@@ -498,6 +499,29 @@ public void appendBehaviorGroupToEventType(final String orgId, final UUID behavi
         }
     }
 
+    @Transactional
+    public void appendActionToBehaviorGroup(final String orgId, final UUID behaviorGroupUuid, final UUID endpointUuid, final int position) {
+        final String appendBehaviorGroupQuery =
+            "INSERT INTO " +
+                "behavior_group_action(behavior_group_id, endpoint_id, created, position) " +
+                "SELECT bg.id, ep.id, :created, :position " +
+                "FROM behavior_group AS bg, endpoints AS ep " +
```

Comment:
This query creates a [cartesian product](https://www.tutorialspoint.com/sql/sql-cross-join.htm) of the `behavior_group` and `endpoints` tables. This would solve the issue:
```sql
INSERT INTO behavior_group_action(behavior_group_id, endpoint_id, created, position)
SELECT :behaviorGroupUuid, :endpointUuid, :created, :position
WHERE EXISTS (SELECT 1 FROM behavior_group bg WHERE bg.id = :behaviorGroupUuid AND bg.org_id = :orgId)
AND EXISTS (SELECT 1 FROM endpoints ep WHERE ep.id = :endpointUuid AND ep.org_id = :orgId)
ON CONFLICT (behavior_group_id, endpoint_id) DO NOTHING
```

Reply:
I refactored this query because it wasn't relevant.
I also added a unit test case for it.